front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useMemo, useState, useEffect } from 'react';
+
+import type { TFunction } from 'i18next';
+import { compact, findLastIndex } from 'lodash';
+import {
+  keyColumn,
+  type Column,
+  checkboxColumn,
+  createTextColumn,
+  DynamicDataSheetGrid,
+} from 'react-datasheet-grid';
+import { useTranslation } from 'react-i18next';
+
+import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
+import { useOsrdConfActions } from 'common/osrdContext';
+import { isVia } from 'modules/pathfinding/utils';
+import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
+import type { PathStep } from 'reducers/osrdconf/types';
+import { useAppDispatch } from 'store';
+
+import timeColumn from './TimeColumnComponent';
+
+type TimesStopsProps = {
+  pathProperties: ManageTrainSchedulePathProperties;
+  pathSteps?: (PathStep | null)[];
+  setPathProperties: (pathProperties: ManageTrainSchedulePathProperties) => void;
+};
+
+type PathWaypointColumn = SuggestedOP & {
+  isMarginValid: boolean;
+};
+
+const marginRegExValidation = /^\d+(\.\d+)?%$|^\d+(\.\d+)?min\/km$/;
```

Comment:
I think your regex is not valid, it on authorize min/km and not min/100km
```suggestion
const marginRegExValidation = /^\d+(\.\d+)?%$|^\d+(\.\d+)?min\/100km$/;
```

Reply:
Done